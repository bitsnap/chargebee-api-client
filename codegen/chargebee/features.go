package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListFeatures generates chargebee client code to fetch all features
//
// API: https://{site}.chargebee.com/api/v2/features
//
// Documentation: https://apidocs.chargebee.com/docs/api/features?lang=curl#list_features
func GenerateListFeatures() string {
	return client.GenerateList("Features", "Feature", "chargebee.com/api/v2/features")
}

// GenerateRetrieveFeature generates chargebee client code to retrieve specific feature
//
// API:   https://{site}.chargebee.com/api/v2/features/{feature-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/features?lang=curl#retrieve_a_feature
func GenerateRetrieveFeature() string {
	return client.GenerateRetrieve("Feature", "chargebee.com/api/v2/features")
}

// GenerateFeaturesModel generates chargebee feature domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/features?lang=curl#feature_attributes
func GenerateFeaturesModel() string {
	return client.GenerateModel("Feature", "Feature attributes", "https://apidocs.chargebee.com/docs/api/features")
}
