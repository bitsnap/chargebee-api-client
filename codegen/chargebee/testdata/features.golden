# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee features code generation when generating Chargebee API client should be able to generate features models" = '''
(string) (len=666) "type Feature struct {\n    Id string `json:\"id\" validate:\"required\"`\n    Name string `json:\"name\" validate:\"required\"`\n    Description string `json:\"description\"`\n    Status enums.StatusEnum `json:\"status\"`\n    Type enums.TypeEnum `json:\"type\"`\n    Unit string `json:\"unit\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    CreatedAt *time.Time `json:\"created_at\" validate:\"required\"`\n    Levels []Level `json:\"levels\"`\n    Name string `json:\"name\"`\n    Value string `json:\"value\" validate:\"required\"`\n    Level int `json:\"level\" validate:\"required\"`\n    IsUnlimited bool `json:\"is_unlimited\" validate:\"required\"`\n}"
'''
